apiserver/facades/client/client/status.go

```diff
@@ -133,12 +133,9 @@ func (c *Client) StatusHistory(request params.StatusHistoryRequests) params.Stat
 			continue
 		}
 
-		var (
-			err  error
-			hist []params.DetailedStatus
-		)
+		var err error
+		var hist []params.DetailedStatus
 		kind := status.HistoryKind(request.Kind)
-		err = errors.NotValidf("%q requires a unit, got %T", kind, request.Tag)
```

Comment:
Removing this line will introduce a bug if the tag is wrong. The code could be better constructed for sure.